src/model/id.rs

```diff
@@ -331,6 +345,61 @@ pub(crate) mod snowflake {
     use serde::de::{Error, Visitor};
     use serde::{Deserializer, Serializer};
 
+    /// impl from std ([T]::as_chunks)
+    const fn as_chunks<T, const N: usize>(slice: &[T]) -> (&[[T; N]], &[T]) {
+        assert!(N != 0, "chunk size must be non-zero");
+        let len = slice.len() / N;
+        let (multiple_of_n, remainder) = slice.split_at(len * N);
+        let new_len = multiple_of_n.len() / N;
+        // SAFETY: We cast a slice of `new_len * N` elements into
+        // a slice of `new_len` many `N` elements chunks.
+        (unsafe { std::slice::from_raw_parts(multiple_of_n.as_ptr().cast(), new_len) }, remainder)
```

Comment:
Is there no way to get rid of this unsafe?

Reply:
This comes from std-- as far as im concerned, this isnt unsafe.

Reply:
i mean, i could try matching over len

Reply:
Or i can add a dependency to bytemuck

Reply:
`bytemuck

Reply:
Really?

Reply:
Its the same thing though...

Reply:
> same thing though

except there are more eyes on the bytemuck unsafe part, and thus if bytemuck gets something fixed, so does serenity - which isn't true if everyone just copy-pastes unsafe stuff that could be a crate

Reply:
i turned it into a crate, [`pieced`](https://docs.rs/pieced)